### PR TITLE
CMakeToolchain refactor

### PR DESCRIPTION
**Pull request by memsharded**
Changelog: Omit
Docs: Omit

This had some level of risk of breaking:
- If users are ``select()`` selecting blocks, but count on the variables + preprocessor definitions to be there. I have explicitly excluded the new "variables" and "preprocessor" blocks to be filtered out by ``select()``, they won't and will need explicit ``.remove()`` them if desired, or use the new ``keep()`` method that supersedes ``select()``

Refactors:
- Adding automatically a header for every block with its name
- Adding explanation to every block about the inputs and definitions
- Moved some global code in the toolchain to its own dedicated blocks
- Added some more ``message()`` information when Conan define values in the toolchain
- Normalizing the blank lines
- Some minor PEP style fixes

The goal is to make the generated file more friendly to read, with more hints of what can be configured, and also to prepare the toolchain for a better customization and selection of what blocks to use.

### CODE REVIEW

**Review by perseoGI — APPROVED** [review]
I really like this PR, removing and trimming whitespaces and refactor code is always great received!

File: conan/tools/cmake/toolchain/blocks.py [hunk 1]
```diff
@@ -1173,6 +1301,16 @@ def remove(self, name, *args):
...
+        """
+        keep the blocks provided as arguments, remove the others, except pre-existing "variables"
+        and "preprocessor", to not break behavior
+        """
+        self._conanfile.output.warning("CMakeToolchain.select is deprecated. Use CMakeToolchain.keep"
+                                       " instead", warn_tag="deprecated")
+        to_keep = [name] + list(args) + ["variables", "preprocessor"]
+        self._blocks = OrderedDict((k, v) for k, v in self._blocks.items() if k in to_keep)
+
+    def keep(self, name, *args):
```
Comment (czoido) [inline conan/tools/cmake/toolchain/blocks.py]: some alternatives for the name:
- filter
- pick
- include
- enable

**Review by czoido — APPROVED** [review]

File: conan/tools/cmake/toolchain/blocks.py (hunk 1 above)
Comment (memsharded) [reply conan/tools/cmake/toolchain/blocks.py]: Something more explicit like ``tc.blocs.enabled(["block1", ...])`` looks good to me.

Future conf would read ``tools.cmake.cmaketoolchain:blocks_enabled=["block1", ]``? or ``enabled_blocks``?

Comment (AbrilRBS) [reply conan/tools/cmake/toolchain/blocks.py]: `enabled_blocks` 👍

**Review by franramirez688 — APPROVED** [review]
👏

File: conan/tools/cmake/toolchain/blocks.py (hunk 1 above)
Comment (memsharded) [reply conan/tools/cmake/toolchain/blocks.py]: Done, renamed to ``blocks.enabled`` (the conf will be another PR)

**Review by AbrilRBS — APPROVED** [review]

**Pull request merged**